internal/pkg/agent/application/info/agent_metadata.go

```diff
@@ -136,18 +138,23 @@ func Metadata() (*ECSMeta, error) {
 }
 
 // ECSMetadata returns an agent ECS compliant metadata.
-func (i *AgentInfo) ECSMetadata() (*ECSMeta, error) {
-	hostname, err := os.Hostname()
-	if err != nil {
-		return nil, err
-	}
-
+func (i *AgentInfo) ECSMetadata(l *logger.Logger) (*ECSMeta, error) {
 	sysInfo, err := sysinfo.Host()
 	if err != nil {
 		return nil, err
 	}
 
 	info := sysInfo.Info()
+	hostname := info.Hostname
+	l.Debugf("in ECSMetadata, hostname = %s", hostname)
```

Comment:
I only added this for debugging purposes, so we could see debug-level logs from this method regardless of whether the FQDN feature flag was set or not.  Will remove before this PR is merged.

Reply:
Forgot to remove this line in my excitement to merge this PR, so here's the follow up PR to remove it: https://github.com/elastic/elastic-agent/pull/2451